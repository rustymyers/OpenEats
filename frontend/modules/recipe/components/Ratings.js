import React from 'react'

require("./../css/ratings.scss");

class Ratings extends React.Component {
  render() {
    let stars = this.props.stars ? this.props.stars : 0;
    if (stars > 5) {
      stars = 5;
    } else if (stars < 0) {
      stars = 0;
    }

    const full_stars = [...Array(stars).keys()].map(function(name, index) {
      return <span key={ index } className="glyphicon glyphicon-star"/>;
    });
    const empty_stars = [...Array(5 - stars).keys()].map(function(name, index) {
      return <span key={ index } className="glyphicon glyphicon-star-empty"/>;
    });

    return (
      <p className="rating-stars">
        { full_stars }
        { empty_stars }
      </p>
    );
  }
}

module.exports = Ratings;
