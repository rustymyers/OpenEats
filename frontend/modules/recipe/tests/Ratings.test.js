import React from 'react';
import Ratings from '../components/Ratings';
import renderer from 'react-test-renderer';

test('2 Star Rating test', () => {
  const component = renderer.create(
    <Ratings stars={ 2 }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('0 Star Rating test', () => {
  const component = renderer.create(
    <Ratings stars={ -3 }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('4 Star Rating test', () => {
  const component = renderer.create(
    <Ratings stars={ 4 }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('5 Star Rating test', () => {
  const component = renderer.create(
    <Ratings stars={ 24 }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
