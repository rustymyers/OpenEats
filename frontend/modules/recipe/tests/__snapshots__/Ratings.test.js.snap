// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`0 Star Rating test 1`] = `
<p
  className="rating-stars"
>
  <span
    className="glyphicon glyphicon-star-empty"
  />
  <span
    className="glyphicon glyphicon-star-empty"
  />
  <span
    className="glyphicon glyphicon-star-empty"
  />
  <span
    className="glyphicon glyphicon-star-empty"
  />
  <span
    className="glyphicon glyphicon-star-empty"
  />
</p>
`;

exports[`2 Star Rating test 1`] = `
<p
  className="rating-stars"
>
  <span
    className="glyphicon glyphicon-star"
  />
  <span
    className="glyphicon glyphicon-star"
  />
  <span
    className="glyphicon glyphicon-star-empty"
  />
  <span
    className="glyphicon glyphicon-star-empty"
  />
  <span
    className="glyphicon glyphicon-star-empty"
  />
</p>
`;

exports[`4 Star Rating test 1`] = `
<p
  className="rating-stars"
>
  <span
    className="glyphicon glyphicon-star"
  />
  <span
    className="glyphicon glyphicon-star"
  />
  <span
    className="glyphicon glyphicon-star"
  />
  <span
    className="glyphicon glyphicon-star"
  />
  <span
    className="glyphicon glyphicon-star-empty"
  />
</p>
`;

exports[`5 Star Rating test 1`] = `
<p
  className="rating-stars"
>
  <span
    className="glyphicon glyphicon-star"
  />
  <span
    className="glyphicon glyphicon-star"
  />
  <span
    className="glyphicon glyphicon-star"
  />
  <span
    className="glyphicon glyphicon-star"
  />
  <span
    className="glyphicon glyphicon-star"
  />
</p>
`;
